merge-recursive.c

```diff
@@ -3808,8 +3809,7 @@ int merge_recursive_generic(struct merge_options *opt,
 	}
 
 	repo_hold_locked_index(opt->repo, &lock, LOCK_DIE_ON_ERROR);
-	clean = merge_recursive(opt, head_commit, next_commit, ca,
-				result);
+	clean = merge_fn(opt, head_commit, next_commit, ca, result);
```

Comment:
It is odd that this doesn't already allow the ORT strategy if configured to do so.

I went hunting for how this was configured in `builtin/merge.c`, but that seems like a very complicated procedure.

We could use something like the `pull.twohead` config here as a way to allow a user to modify the merge strategy. That gives us a safety valve in case the ORT strategy has a problem in one of these consumers.

The other approach could be to just use the ORT algorithm here no matter what. It's a big change, but it might improve things more rapidly.

After having _all of these thoughts_ I think you've found a good middle ground: the function pointer presents a way to use ORT only when we really need it. This works especially well for our early adoption of sparse index in `microsoft/git`, but we should call special attention to this when upstreaming the work.